File: Haskell/beginner/1958.hs
-- https://judge.beecrowd.com/pt/problems/view/1958
-- Notação Científica

import Text.Printf (printf)

-- Helper function to ensure the correct exponent format (E+00, E-07, etc.)
formatScientificBeecrowd :: Double -> String
formatScientificBeecrowd 0.0 = "+0.0000E+00"
formatScientificBeecrowd n =
    let s = printf "%.4E" n :: String
        (mantissa, rest) = break (== 'E') s
        exponentPart = drop 1 rest -- Remove 'E'

        -- Ensure mantissa starts with '+' if positive, otherwise keep as is.
        finalMantissa = if head mantissa /= '-' then '+' : mantissa else mantissa

        -- Format exponent to always have two digits after the sign (e.g., "+00", "-07", "+12").
        -- The 'printf' already gives "+0" or "-12", etc.
        -- We need to handle cases like "+0" -> "+00", "-7" -> "-07".
        formattedExp = case exponentPart of
            [s, d] | s == '+' || s == '-' -> [s, '0', d] -- e.g., "+0" -> "+00", "-7" -> "-07"
            _ -> exponentPart -- Already two digits or more, e.g., "+12", "-10"
    in
    finalMantissa ++ "E" ++ formattedExp

main :: IO ()
main = do
    usrInput <- readLn :: IO Double
    putStrLn $ formatScientificBeecrowd usrInput
-- By: Fernando Serra
-- https://github.com/fernandosserra

File: Python/beginner/1958.py
# https://judge.beecrowd.com/pt/problems/view/1958
# Notação Científica

# Main
usrInput = float(input())


formatted_str = f"{usrInput:+.4E}"

e_pos = formatted_str.find('E')
mantissa = formatted_str[:e_pos]
exponent_part = formatted_str[e_pos+1:]
    
if len(exponent_part) == 2 and (exponent_part[0] == '+' or exponent_part[0] == '-'):       
    formatted_exponent = exponent_part[0] + '0' + exponent_part[1]
elif len(exponent_part) == 1 and exponent_part[0].isdigit():
    formatted_exponent = "+0" + exponent_part[0]
else:
    formatted_exponent = exponent_part
    
print(f"{mantissa}E{formatted_exponent}")

# By: Fernando Serra
# https://github.com/fernandosserra